python/lsst/ci/hsc/gen2/validate.py

```diff
@@ -228,6 +229,27 @@ def validateMatchFull(self, dataId):
         matches = self.butler.get(self._matchFullDataset, dataId)
         self.assertGreater("Number of full matches", len(matches), self._minMatches)
 
+    def validateSchema(self, dataset, dataId, tableName):
+        """Check the schema of the parquet dataset match that in the DDL"""
+        self.log.info("Validating %s match the schema in %s", dataset, self.filepath)
+        with open(self.filepath, 'r') as f:
+            tables = yaml.safe_load(f)['tables']
+        sdmSchema = [table for table in tables if table['name'] == tableName]
+        self.assertEqual("There should be just one DDL for this table", len(sdmSchema), 1)
+        expectedColumnNames = set(column['name'] for column in sdmSchema[0]['columns'])
+
+        outputTable = self.butler.get(dataset, dataId)
+        # The type of outputTable is lsst.pipe.tasks.parquetTable.ParquetTable
+        # in Gen2, but is pandas.DataFrame in Gen3.
+        if isinstance(outputTable, ParquetTable):
+            df = outputTable.toDataFrame()
```

Comment:
`outputTable.columns` gives you the column list without reading in the entire table

Reply:
I forget what the corresponding gen3-ism is

Reply:
The problem is... for Gen2 `objectTable` `ParquetTable`, `outputTable.columns` is different from `outputTable.toDataFrame().columns.to_list()`. Somehow the former has an extra item "id".